x-pack/plugins/aiops/server/routes/log_rate_analysis/queries/fetch_top_terms.ts

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { uniqBy } from 'lodash';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ElasticsearchClient } from '@kbn/core/server';
+
+import type { Logger } from '@kbn/logging';
+import { type SignificantItem, SIGNIFICANT_ITEM_TYPE } from '@kbn/ml-agg-utils';
+import {
+  createRandomSamplerWrapper,
+  type RandomSamplerWrapper,
+} from '@kbn/ml-random-sampler-utils';
+
+import { RANDOM_SAMPLER_SEED } from '../../../../common/constants';
+import type { AiopsLogRateAnalysisSchema } from '../../../../common/api/log_rate_analysis/schema';
+
+import { isRequestAbortedError } from '../../../lib/is_request_aborted_error';
+
+import { getQueryWithParams } from './get_query_with_params';
+import { getRequestBase } from './get_request_base';
+
+// TODO Consolidate with duplicate `fetchDurationFieldCandidates` in
```

Comment:
Is this comment still needed?

Reply:
Yes, this is an older comment I copied over, it's related to this issue where we track potential cleanups of duplicated code: https://github.com/elastic/kibana/issues/136182